pkg/diagnostics/http_tracing.go

```diff
@@ -123,43 +119,25 @@ func isHealthzRequest(name string) bool {
 // UpdateSpanStatusFromHTTPStatus updates trace span status based on response code.
 func UpdateSpanStatusFromHTTPStatus(span *trace.Span, code int) {
 	if span != nil {
-		span.SetStatus(traceStatusFromHTTPCode(code))
+		statusCode, statusDescription := traceStatusFromHTTPCode(code)
+		(*span).SetStatus(statusCode, statusDescription)
 	}
 }
 
 // https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#status
-func traceStatusFromHTTPCode(httpCode int) trace.Status {
-	code := codes.Unknown
-	switch httpCode {
-	case http.StatusUnauthorized:
-		code = codes.Unauthenticated
-	case http.StatusForbidden:
-		code = codes.PermissionDenied
-	case http.StatusNotFound:
-		code = codes.NotFound
-	case http.StatusTooManyRequests:
-		code = codes.ResourceExhausted
-	case http.StatusNotImplemented:
-		code = codes.Unimplemented
-	case http.StatusServiceUnavailable:
-		code = codes.Unavailable
-	case http.StatusGatewayTimeout:
-		code = codes.DeadlineExceeded
-	}
-
-	if code == codes.Unknown {
-		if httpCode >= 100 && httpCode < 300 {
-			code = codes.OK
-		} else if httpCode >= 300 && httpCode < 400 {
-			code = codes.DeadlineExceeded
-		} else if httpCode >= 400 && httpCode < 500 {
-			code = codes.InvalidArgument
-		} else if httpCode >= 500 {
-			code = codes.Internal
+func traceStatusFromHTTPCode(httpCode int) (otelcodes.Code, string) {
```

Comment:
Is this a breaking change?

Reply:
It could be indeed a breaking change. I just tried to apply the logic referenced in the function documentation. See: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#status
Not sure previous implementation was actually aligned with the opentelemetry convention.